salt/states/pkg.py

```diff
@@ -1624,18 +1625,18 @@ def latest(
 
     targets = {}
     problems = []
+    cmp_func = __salt__.get('pkg.version_cmp')
     for pkg in desired_pkgs:
```

Comment:
The changes below are based on an incorrect assumption of how ``pkg.latest_version`` is supposed to work, likely due to there being a bug in pacman.py which I have fixed in https://github.com/saltstack/salt/pull/35880.

There is no need to do a version comparison here, this was deliberately removed recently and should not be added back. If ``avail[pkg]`` evaluates as false, then there is either a) no upgrade available, or b) a package by that name does not exist.

If ``avail[pkg]`` evaluates as true, then there _is_ an update available and the package should be added to the installation targets. The mere fact that ``avail[pkg]`` contains something other than an empty string is enough to determine that the package needs to be updated, there is no need for a version comparison (which is why it was removed).

I added some clarifying comments to this block of code in https://github.com/saltstack/salt/pull/35882, as well as fixed the condition for Gentoo USE flags.